compose/compiler/compiler-hosted/integration-tests/src/test/java/androidx/compose/compiler/plugins/kotlin/ControlFlowTransformTests.kt

```diff
@@ -44,14 +44,55 @@ class ControlFlowTransformTests : AbstractControlFlowTransformTests() {
         """
     )
 
+    @Test
+    fun testAND(): Unit = controlFlow(
+        """
+            @NonRestartableComposable
+            @Composable
+            fun Example() {
+                B() && B()
+            }
+        """,
+        """
+            @NonRestartableComposable
+            @Composable
+            fun Example(%composer: Composer?, %changed: Int) {
+              %composer.startReplaceableGroup(<>, "C(Example)<B()>,<B()>:Test.kt")
+              val tmp0_group = B(%composer, 0) && B(%composer, 0)
+              tmp0_group
+              %composer.endReplaceableGroup()
+            }
+        """
+    )
+
+    @Test
+    fun testOR(): Unit = controlFlow(
+        """
+            @NonRestartableComposable
+            @Composable
+            fun Example() {
+                B() || B()
+            }
+        """,
+        """
+            @NonRestartableComposable
+            @Composable
+            fun Example(%composer: Composer?, %changed: Int) {
+              %composer.startReplaceableGroup(<>, "C(Example)<B()>,<B()>:Test.kt")
+              val tmp0_group = B(%composer, 0) || B(%composer, 0)
+              tmp0_group
+              %composer.endReplaceableGroup()
+            }
+        """
+    )
+
     @Test
     fun testIfWithCallsInBranch(): Unit = controlFlow(
         """
             @NonRestartableComposable @Composable
             fun Example(x: Int) {
-                // Composable calls in the result blocks, so we can determine static number of
-                // groups executed. This means we put a group around the "then" and the implicit
-                // "else" blocks
+                // Only one composable call in the result blocks, so we can just generate 
+                // a single group around the whole expression. 
```

Comment:
This line has trailing whitespace, which triggers a lint warning in our CI. Could you please remove it? Sorry, I didn't catch this earlier.